src/wms/src/main/java/org/geoserver/wms/capabilities/Capabilities_1_3_0_Transformer.java

```diff
@@ -121,19 +122,43 @@
 
     private WMS wmsConfig;
 
+    /** whether to skip root Layer element if there is a already single top Layer element * */
+    private Boolean noRootLayer = null;
+
     /**
      * Creates a new WMSCapsTransformer object.
      *
      * @param wms
      * @param schemaBaseUrl the base URL of the current request (usually
      *     "http://host:port/geoserver")
      * @param getMapFormats the list of supported output formats to state for the GetMap request
+     * @param extCapsProviders collection of providers of extended capabilities content
      */
     public Capabilities_1_3_0_Transformer(
             WMS wms,
             String schemaBaseUrl,
             Collection<GetMapOutputFormat> getMapFormats,
             Collection<ExtendedCapabilitiesProvider> extCapsProviders) {
+        this(wms, schemaBaseUrl, getMapFormats, extCapsProviders, null);
+    }
+
+    /**
+     * Creates a new WMSCapsTransformer object.
+     *
+     * @param wms
+     * @param schemaBaseUrl the base URL of the current request (usually
+     *     "http://host:port/geoserver")
+     * @param getMapFormats the list of supported output formats to state for the GetMap request
+     * @param extCapsProviders collection of providers of extended capabilities content
+     * @param noRootLayer whether to skip root Layer element if there is a already single top Layer
+     *     element
+     */
+    public Capabilities_1_3_0_Transformer(
+            WMS wms,
+            String schemaBaseUrl,
+            Collection<GetMapOutputFormat> getMapFormats,
+            Collection<ExtendedCapabilitiesProvider> extCapsProviders,
+            Boolean noRootLayer) {
```

Comment:
This new optional parameter caused many changes in the code, with nulls added at the end of various constructor calls... wouldn't it be better to make it a setter and invoke it only when needed instead, leaving the constructor as it was?